solr/core/src/test/org/apache/solr/search/ReturnFieldsTest.java

```diff
@@ -395,6 +395,16 @@ public void testFunkyFieldNames() {
         "//arr[@name='#foo_s']/str[.='how now brown cow']");
   }
 
+  @Test
+  public void testRowsZeroNoScore() {
+    ReturnFields rf = new SolrReturnFields(req("fl", "id,score", "rows", "0"));
+    assertFalse(rf.wantsScore());
+    assertFalse(rf.wantsField("score"));
+    assertTrue(rf.wantsField("id"));
```

Comment:
I question that this makes sense.  If we're going to remove "score" then why not everything really?  There are no rows, after all.

Reply:
Yea.... so I had the same thought just remove all fields if no rows, but that caused a whole bunch of test failures. I agree with what you are saying. I can look at it again just to make sure.

Reply:
A TODO is fine.